### ISSUE TITLE
Feature request: Add "test name" option to ValidTestCase / InvalidTestCase

### ISSUE THREAD

**Issue by ilyub** [typescript-eslint/typescript-eslint#3920]
**Repro**

```TS
ruleTester.run("my-rule", rule, {
    invalid: [
        {
            name: "Test name", <---------------------- New option
            code: "foo",
            errors: [
                {
                    messageId: "avoidName"
                }
            ]
        }
    ]
});
```

**Expected Result**

When there are many tests it is easier to find failed test by name than by code. So, I think name option will make it easier to work with tests.

**Actual Result**

-

**Additional Info**

-
